miner/bid_simulator.go

```diff
@@ -359,21 +389,34 @@ func (b *bidSimulator) newBidLoop() {
 			}
 
 			var replyErr error
-			// simulatingBid will be nil if there is no bid in simulation, compare with the bestBid instead
-			if simulatingBid := b.GetSimulatingBid(newBid.bid.ParentHash); simulatingBid != nil {
-				// simulatingBid always better than bestBid, so only compare with simulatingBid if a simulatingBid exists
-				if bidRuntime.isExpectedBetterThan(simulatingBid) {
-					commit(commitInterruptBetterBid, bidRuntime)
+			if preBestBidToRun, exist := b.GetBestBidToRun(newBid.bid.ParentHash); exist {
```

Comment:
for the first bid, exist will be false,
so SetBestBidToRun wiil never be called, 
so GetBestBidToRun will always return false